changelog/unreleased/pull-2718

```diff
@@ -0,0 +1,17 @@
+Enhancement: Improve pruning performance and make pruning more customizable
+
+The prune command is now much faster. This is especially the case for remote
+repositories or repositories with not much to prune.
+Also the memory usage of the prune command is now reduced.
+
+The prune command now also supports new user flags, see
+https://restic.readthedocs.io/en/stable/060_forget.html
+
+Fixes several open issues, e.g.:
+https://github.com/restic/restic/issues/1599
```

Comment:
Completely solving #1599 requires getting rid of both reindex steps.

Reply:
This is now removed. I think it can be added again by #2842.